pallets/creator-staking/src/types/impls.rs

```diff
@@ -0,0 +1,273 @@
+        )
+    }
+}
+
+impl<Balance> UnbondingInfo<Balance>
+    where
+        Balance: AtLeast32BitUnsigned + Default + Copy + MaxEncodedLen,
+{
+    /// Returns total number of unlocking chunks.
+    pub(crate) fn len(&self) -> u32 {
+        self.unlocking_chunks.len() as u32
+    }
+
+    /// True if no unlocking chunks exist, false otherwise.
+    fn is_empty(&self) -> bool {
+        self.unlocking_chunks.is_empty()
+    }
+
+    /// Returns sum of all unlocking chunks.
+    pub(crate) fn sum(&self) -> Balance {
+        self.unlocking_chunks
+            .iter()
+            .map(|chunk| chunk.amount)
+            .reduce(|c1, c2| c1 + c2)
+            .unwrap_or_default()
+    }
+
+    /// Adds a new unlocking chunk to the vector, preserving the unlock_era based ordering.
+    pub(crate) fn add(&mut self, chunk: UnlockingChunk<Balance>) {
+        // It is possible that the unbonding period changes so we need to account for that
+        match self.unlocking_chunks.binary_search_by(|x| x.unlock_era.cmp(&chunk.unlock_era)) {
+            // Merge with existing chunk if unlock_eras match
+            Ok(pos) => self.unlocking_chunks[pos].add_amount(chunk.amount),
+            // Otherwise insert where it should go. Note that this will in almost all cases return
+            // the last index.
+            Err(pos) => self.unlocking_chunks.insert(pos, chunk),
+        }
+    }
+
+    /// Partitions the unlocking chunks into two groups:
+    ///
+    /// First group includes all chunks which have unlock era lesser or equal to the specified era.
+    /// Second group includes all the rest.
+    ///
+    /// Order of chunks is preserved in the two new structs.
+    pub(crate) fn partition(self, era: EraIndex) -> (Self, Self) {
+        let (matching_chunks, other_chunks): (
+            Vec<UnlockingChunk<Balance>>,
+            Vec<UnlockingChunk<Balance>>,
+        ) = self.unlocking_chunks.iter().partition(|chunk| chunk.unlock_era <= era);
+
+        (Self { unlocking_chunks: matching_chunks }, Self { unlocking_chunks: other_chunks })
+    }
+}
+
+impl Default for RewardDistributionConfig {
+    /// `default` values based on configuration at the time of writing this code.
+    /// Should be overriden by desired params.
+    fn default() -> Self {
+        RewardDistributionConfig {
+            stakers_percent: Perbill::from_percent(50),
+            creators_percent: Perbill::from_percent(50),
+        }
+    }
+}
+
+impl RewardDistributionConfig {
+    /// `true` if sum of all percentages is `one whole`, `false` otherwise.
+    pub fn is_consistent(&self) -> bool {
+        let variables = vec![
+            &self.stakers_percent,
+            &self.creators_percent,
+        ];
+        let accumulator = variables
+            .iter()
+            .try_fold(Perbill::zero(), |acc, &percent| acc.checked_add(percent))
+            .unwrap_or_default();
+
+        Perbill::one() == accumulator
```

Comment:
Can we rewrite it in a simple way? Something like:
`Perbill::one() == (self.stakers_percent + &self.creators_percent)`

Reply:
As soon as we'll add more than 2 options there, we'll need to get that fold back.